arch/x86/core/early_serial.c

```diff
@@ -11,10 +11,10 @@
 #include <soc.h>
 
 
-#ifdef UART_NS16550_ACCESS_IOPORT
+#ifdef CONFIG_UART_NS16550_ACCESS_IOPORT
 /* Legacy I/O Port Access to a NS16550 UART */
-#define IN(reg)       sys_in8(reg + UART_NS16550_ACCESS_IOPORT)
-#define OUT(reg, val) sys_out8(val, reg + UART_NS16550_ACCESS_IOPORT)
+#define IN(reg)       sys_in8(reg + DT_REG_ADDR(DT_CHOSEN(zephyr_console)))
+#define OUT(reg, val) sys_out8(val, reg + DT_REG_ADDR(DT_CHOSEN(zephyr_console)))
```

Comment:
This code continues to go in circles.  The reason "early serial" doesn't use DTS is because this code tends to be brought up on new platforms before a working DTS configuration is present.

I've mostly given up on the idea that this can be maintained as a standalone driver that can be easily mutated during bringup, it's just too tempting to try to tie it into everything else.  So no -1, but...